pkg/gits/github.go

```diff
@@ -325,17 +325,18 @@ func (p *GitHubProvider) ForkRepository(originalOrg string, name string, destina
 		if strings.Contains(err.Error(), "try again later") {
 			log.Warnf("Waiting for the fork of %s/%s to appear...\n", owner, name)
 			// lets wait for the fork to occur...
-			start := time.Now()
-			deadline := start.Add(time.Minute)
+			timer := time.NewTimer(time.Minute)
+			defer timer.Stop()
 			for {
-				time.Sleep(5 * time.Second)
-				repo, _, err = p.Client.Repositories.Get(p.Context, owner, name)
-				if repo != nil && err == nil {
-					break
-				}
-				t := time.Now()
-				if t.After(deadline) {
+				select {
+				case <-timer.C:
 					return nil, fmt.Errorf("Gave up waiting for Repository %s/%s to appear: %s", owner, name, err)
+				default:
+					time.Sleep(5 * time.Second)
+					repo, _, err = p.Client.Repositories.Get(p.Context, owner, name)
```

Comment:
We should definitely use `context.WithTimeout` instead to cancel the request when timeout expires.

https://golang.org/pkg/context/#WithTimeout